backend/helpers/emailService.js

```diff
@@ -0,0 +1,39 @@
+const nodemailer = require('nodemailer');
+const to = require('await-to-js').default;
+
+const { ErrorHandler } = require('./error');
+const constants = require('../constants');
+const { EMAIL_USER, EMAIL_PASS, EMAIL_HOST } = require('../config/index');
+const { getMailTemplate } = require('../utility/emailTemplates');
+
+const transporter = nodemailer.createTransport({
+  type: 'SMTP',
+  host: EMAIL_HOST,
+  secure: true,
+  debug: true,
+  port: 465,
+  auth: {
+    user: EMAIL_USER,
+    pass: EMAIL_PASS,
+  },
+});
+
+module.exports.sendEmail = async (email, data, type) => {
+  const { subject, text } = getMailTemplate(type, data);
+  const mailOptions = {
+    from: EMAIL_USER,
+    to: email,
+    subject,
+    text,
+  };
+  const [err, response] = await to(transporter.sendMail(mailOptions));
+  if (err) {
+    const error = new ErrorHandler(constants.ERRORS.UNEXPECTED, {
+      statusCode: '500',
+      message: 'The server encountered an unexpected condition which prevented it from fulfilling the request.',
+      errorStack: err,
```

Comment:
I cannot understand where the user id will be during the api call.
I could find res.locals.decode.email in a few controllers.

Reply:
In this case the user can be the email field. 
res.locals.decode is used when it is an authenticated API

Reply:
The email field would have the recipients email, is that suitable here?

Reply:
Yes, ik it isn't 100% suitable but it might help later in retracing the error and probably once when we proper logger and nested error handling like situations